src/components/ModalViews/ReviewTransactionView.tsx

```diff
@@ -71,7 +117,12 @@ export const ReviewTransactionView: React.FC<ReviewTransactionViewProps> = (prop
         <ModalLoadingGate />
       </>
     )
-  } else if (!isIdle && !!amountUnformatted && allowanceUnformatted?.lt(amountUnformatted)) {
+  } else if (
+    !isIdle &&
+    !!amountUnformatted &&
+    allowanceUnformatted?.lt(amountUnformatted) &&
+    !hasAllValidApprovalSignatures()
```

Comment:
Should this check only run if `approvalType` is signatures? Seems like it might cause problems otherwise.